public/js/controllers/attackcost_controller.js

```diff
@@ -0,0 +1,385 @@
+import { Controller } from 'stimulus'
+import TurboQuery from '../helpers/turbolinks_helper'
+import { getDefault } from '../helpers/module_helper'
+import globalEventBus from '../services/event_bus_service'
+import { nightModeOptions } from '../helpers/chart_helper'
+import dompurify from 'dompurify'
+
+function digitformat (amount, decimalPlaces, noComma) {
+  if (!amount) return 0
+
+  if (noComma) return amount.toFixed(decimalPlaces)
+
+  decimalPlaces = decimalPlaces || 0
+  let result = parseFloat(amount).toLocaleString(undefined, { minimumFractionDigits: decimalPlaces, maximumFractionDigits: decimalPlaces }).replace(/\.0*$/, '')
```

Comment:
Better to define these string formatter objects as constants than to recreate them every time. Consider using sig figs too.

```
const legendFormatter = {
  minimumSignificantDigits: 4,
  maximumSignificantDigits: 5
}
```

Or consider using `humanize.threeSigFigs` which will scale the value and append an appropriate 'k, M, B' unit prefix.

Reply:
I can't see how this can be made a constant as it used with varying parameters all through the controller